pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2041,10 +2018,10 @@ func getActualWildcardAndLocalHost(node *model.Proxy) (string, string) {
 }
 
 // getSidecarInboundBindIP returns the IP that the proxy can bind to along with the sidecar specified port.
-// It looks for an unicast address, if none found, then the default wildcard address is used.
-// This will make the inbound listener bind to instance_ip:port instead of 0.0.0.0:port where applicable.
+// It looks for an unicast address, if none found, then the default loopback address is used.
+// This will make the inbound listener bind to instance_ip:port instead of 127.0.0.1:port where applicable.
 func getSidecarInboundBindIP(node *model.Proxy) string {
-	defaultInboundIP, _ := getActualWildcardAndLocalHost(node)
+	_, defaultInboundIP := getActualWildcardAndLocalHost(node)
```

Comment:
why this change?

Reply:
See the comment on top of the call site. This was a bug that was introduced. Whether or not bind to port is true, for inbound, we always try to listen on the pod's IP (or VM's IP) if the user has not mentioned any bind IP. If we cannot determine a bind IP, then we fallback to loopback address

Reply:
actually ignore this piece.